clients/client_service/stream.go

```diff
@@ -0,0 +1,82 @@
+package clientservice
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/harness/ff-proxy/v2/domain"
+	"github.com/harness/ff-proxy/v2/log"
+	"github.com/harness/ff-proxy/v2/stream"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/r3labs/sse/v2"
+)
+
+type messageHandler interface {
+	HandleMessage(m domain.SSEMessage) error
+}
+
+// Stream is the type that subscribes to stream that relays Proxy events and handles events coming off the stream
+type Stream struct {
+	log            log.Logger
+	subscriber     stream.Subscriber
+	messageHandler messageHandler
+}
+
+// NewStream opens a subscription to the client service's stream endpoint
+func NewStream(l log.Logger, s stream.Subscriber, m messageHandler) Stream {
+	l = l.With("component", "Start")
+	return Stream{
+		log:            l,
+		subscriber:     s,
+		messageHandler: m,
+	}
+}
+
+// Start connects to the stream and registers a handler to handle events coming off the stream.
+// If the stream disconnects backoff and attempt to subscribe again.
+func (s Stream) Start(ctx context.Context) {
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				err := s.subscriber.Sub(ctx, "*", "", func(id string, v interface{}) error {
+					msg, err := parseMessage(v)
+					if err != nil {
+						return nil
```

Comment:
Are we returning nil here, to avoid the message being re-published?
Should we include any kind of log to indicate what has happened?